addons/mail/static/src/models/thread_view_topbar/thread_view_topbar.js

```diff
@@ -289,6 +372,23 @@ function factory(dependencies) {
         // Private
         //----------------------------------------------------------------------
 
+        /**
+         * @private
+         */
+        _applyGuestRename() {
+            this.update({ isEditingGuestName: false });
+            if (this.pendingGuestName === '') {
+                this._discardGuestRename();
+                return;
+            }
```

Comment:
Confusing method names, something called `_applyGuestRename` should not call `_discardGuestRename` instead as they look opposite by name.